src/Symfony/Bridge/Doctrine/Types/AbstractUidType.php

```diff
@@ -92,4 +95,14 @@ public function requiresSQLCommentHint(AbstractPlatform $platform): bool
     {
         return true;
     }
+
+    private function hasNativeGuidType(AbstractPlatform $platform): bool
+    {
+        // DBAL 2 compatibility
```

Comment:
this is actually compat for DBAL `<3.4`. DBAL 3.3 does not have `getStringTypeDeclarationSQL` either.